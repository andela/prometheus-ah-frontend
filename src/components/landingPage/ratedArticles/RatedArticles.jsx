import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ReactHtmlParser from 'react-html-parser';
import RatedArticle from './RatedArticle';
import ResetPassword from '../../ResetLink';

/**
 * @class RatedArticles
 */
class RatedArticles extends Component {
  /**
   * @description Render the JSX template
   *
   * @memberof LoginModal
   *
   * @returns {JSX} JSX representation of component
   */
  render() {
    const { posts } = this.props;
    return (
      <div>
        <div id="carouselIndicators" className="carousel slide" data-ride="carousel">
          <ResetPassword />
          <div className="d-flex flex-column">
            <div className="landing-label">TOP-RATED</div>
            <ol className="carousel-indicators">
              <li data-target="#carouselIndicators" data-slide-to="0" className="active" />
              <li data-target="#carouselIndicators" data-slide-to="1" />
              <li data-target="#carouselIndicators" data-slide-to="2" />
              <li data-target="#carouselIndicators" data-slide-to="3" />
              <li data-target="#carouselIndicators" data-slide-to="4" />
            </ol>
            <div className="carousel-inner">
              {
                posts.map((post, index) => (
                  <RatedArticle
                    key={post.id}
                    author={post.User.username}
                    title={post.title}
                    body={ReactHtmlParser(post.body)}
                    date={post.createdAt}
                    active={index}
                  />))
              }
            </div>
          </div>
        </div>
      </div>
    );
  }
}

RatedArticles.propTypes = {
  posts: PropTypes.instanceOf(Array)
};

export default RatedArticles;
