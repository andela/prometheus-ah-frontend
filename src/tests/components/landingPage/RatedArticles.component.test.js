import React from 'react';
import { shallow } from 'enzyme';
import RatedArticles from '../../../components/landingPage/ratedArticles/RatedArticles';
import mockData from '../../__mocks__/mockData';

let wrapper;

describe('RatedArticles component', () => {
  beforeEach(() => {
    const { posts } = mockData;
    const props = {
      posts
    };

    wrapper = shallow(<RatedArticles {...props} />);
  });

  test('should render a snapshot of the RatedArticles component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
