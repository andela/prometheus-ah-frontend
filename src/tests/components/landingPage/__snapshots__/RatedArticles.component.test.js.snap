// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatedArticles component should render a snapshot of the RatedArticles component 1`] = `
<div>
  <div
    className="carousel slide"
    data-ride="carousel"
    id="carouselIndicators"
  >
    <Connect(ResetPassword) />
    <div
      className="d-flex flex-column"
    >
      <div
        className="landing-label"
      >
        TOP-RATED
      </div>
      <ol
        className="carousel-indicators"
      >
        <li
          className="active"
          data-slide-to="0"
          data-target="#carouselIndicators"
        />
        <li
          data-slide-to="1"
          data-target="#carouselIndicators"
        />
        <li
          data-slide-to="2"
          data-target="#carouselIndicators"
        />
        <li
          data-slide-to="3"
          data-target="#carouselIndicators"
        />
        <li
          data-slide-to="4"
          data-target="#carouselIndicators"
        />
      </ol>
      <div
        className="carousel-inner"
      >
        <RatedArticle
          active={0}
          author="Joeeasy"
          body={
            Array [
              "My task is getting bigger and I am equal to the task.
        Music gives a soul to the universe, wings to the mind,
        flight to the imagination and life to everything",
            ]
          }
          date="2018-09-20T13:43:12.129Z"
          key="1"
          title="My Broken Mother, My Broken Heart"
        />
        <RatedArticle
          active={1}
          author="Joeeasy"
          body={
            Array [
              "My task is getting bigger and I am equal to the task.
      Music gives a soul to the universe, wings to the mind,
      flight to the imagination and life to everything",
            ]
          }
          date="2018-09-20T13:43:12.129Z"
          key="1"
          title="My Broken Mother, My Broken Heart"
        />
      </div>
    </div>
  </div>
</div>
`;
